pkg/backup/util/util.go

```diff
@@ -148,30 +164,34 @@ func GenerateGcsCertEnvVar(gcs *v1alpha1.GcsStorageProvider) ([]corev1.EnvVar, s
 }
 
 // GenerateStorageCertEnv generate the env info in order to access backend backup storage
-func GenerateStorageCertEnv(ns string, provider v1alpha1.StorageProvider, secretLister corelisters.SecretLister) ([]corev1.EnvVar, string, error) {
+func GenerateStorageCertEnv(ns string, useKMS bool, provider v1alpha1.StorageProvider, secretLister corelisters.SecretLister) ([]corev1.EnvVar, string, error) {
 	var certEnv []corev1.EnvVar
 	var reason string
+	var err error
 	storageType := GetStorageType(provider)
 
 	switch storageType {
 	case v1alpha1.BackupStorageTypeS3:
 		if provider.S3 == nil {
 			return certEnv, "S3ConfigIsEmpty", errors.New("s3 config is empty")
 		}
+
 		s3SecretName := provider.S3.SecretName
-		secret, err := secretLister.Secrets(ns).Get(s3SecretName)
-		if err != nil {
-			err := fmt.Errorf("get s3 secret %s/%s failed, err: %v", ns, s3SecretName, err)
-			return certEnv, "GetS3SecretFailed", err
-		}
+		if s3SecretName != "" {
+			secret, err := secretLister.Secrets(ns).Get(s3SecretName)
+			if err != nil {
+				err := fmt.Errorf("get s3 secret %s/%s failed, err: %v", ns, s3SecretName, err)
+				return certEnv, "GetS3SecretFailed", err
+			}
 
-		keyStr, exist := CheckAllKeysExistInSecret(secret, constants.S3AccessKey, constants.S3SecretKey)
-		if !exist {
-			err := fmt.Errorf("s3 secret %s/%s missing some keys %s", ns, s3SecretName, keyStr)
-			return certEnv, "s3KeyNotExist", err
+			keyStr, exist := CheckAllKeysExistInSecret(secret, constants.S3AccessKey, constants.S3SecretKey)
+			if !exist {
+				err := fmt.Errorf("s3 secret %s/%s missing some keys %s", ns, s3SecretName, keyStr)
+				return certEnv, "s3KeyNotExist", err
+			}
```

Comment:
Now I think it is better to put this check logic into function `GenerateS3CertEnvVar`

Reply:
I think it will be OK here , can we move it after we have to do it ?